// skeleton-elements/angular/esm2015/lib/skeleton-elements.module.js
import { NgModule } from '@angular/core';
import { SkeletonBlockComponent } from './skeleton-block';
import { SkeletonAvatarComponent } from './skeleton-avatar';
import { CommonModule } from '@angular/common';
import { SkeletonTextDirective } from './skeleton-text';
import { SkeletonImageComponent } from './skeleton-image';
import * as i0 from "@angular/core";
export class SkeletonElementsModule {
}
SkeletonElementsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: SkeletonElementsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SkeletonElementsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: SkeletonElementsModule, declarations: [SkeletonBlockComponent,
        SkeletonAvatarComponent,
        SkeletonTextDirective,
        SkeletonImageComponent], imports: [CommonModule], exports: [SkeletonBlockComponent,
        SkeletonAvatarComponent,
        SkeletonTextDirective,
        SkeletonImageComponent] });
SkeletonElementsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: SkeletonElementsModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.1.1", ngImport: i0, type: SkeletonElementsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SkeletonBlockComponent,
                        SkeletonAvatarComponent,
                        SkeletonTextDirective,
                        SkeletonImageComponent,
                    ],
                    exports: [
                        SkeletonBlockComponent,
                        SkeletonAvatarComponent,
                        SkeletonTextDirective,
                        SkeletonImageComponent,
                    ],
                    imports: [CommonModule],
                }]
        }] });
